internal/service/cluster/collector/collector_alarms.go

```diff
@@ -99,17 +99,17 @@ func nodeClusterTypesWithAlarmDictionaryID(nodeClusterTypes []models.NodeCluster
 			continue
 		}
 
-		alarmDictionaryIDString := (*nodeClusterType.Extensions)[utils.ClusterAlarmDictionaryIDExtension]
-		if alarmDictionaryIDString != nil {
-			_, err := uuid.Parse(alarmDictionaryIDString.(string))
+		if alarmDictionaryIDString, ok := (*nodeClusterType.Extensions)[utils.ClusterAlarmDictionaryIDExtension].(string); ok {
+			_, err := uuid.Parse(alarmDictionaryIDString)
 			if err != nil {
 				slog.Error("error parsing alarm dictionary ID", "NodeClusterType ID", nodeClusterType.NodeClusterTypeID, "error", err)
 				continue
 			}
-
-			filteredNodeClusterTypes = append(filteredNodeClusterTypes, nodeClusterType)
 		}
+
+		filteredNodeClusterTypes = append(filteredNodeClusterTypes, nodeClusterType)
```

Comment:
Shouldn't this be inside the previous `if` given that it supposed to be filtering the list down to only those with a valid extension?

Reply:
reverted